Core/Core/API/API.swift

```diff
@@ -101,7 +101,8 @@ public struct URLSessionAPI: API {
     @discardableResult
     public func uploadTask<R: APIRequestable>(_ requestable: R) throws -> URLSessionTask {
         let request = try requestable.urlRequest(relativeTo: baseURL, accessToken: accessToken, actAsUserID: actAsUserID)
-        let url = URL.temporaryDirectory.appendingPathComponent(UUID.string)
+        let directory = urlSession.configuration.sharedContainerIdentifier.flatMap(URL.sharedContainer) ?? URL.temporaryDirectory
```

Comment:
since we don't always write to a temp file, should we make sure to delete the local file when we're done?

Reply:
Done.